packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js

```diff
@@ -593,6 +661,35 @@ const Conversation = WebexPlugin.extend({
       });
   },
 
+  /**
+   * Get url from convo object. If there isn't one, get it from the cache
+   *
+   * @param {Object} convo convo object with either a url or id
+   * @returns {String} url for the specific convo
+   */
+  getConvoUrl({id, url}) {
+    if (url) {
+      if (id) {
+        idToUrl.set(id, url);
```

Comment:
Kind of weird naming on `getConvoUrl` when it's also a setter?

Reply:
Updating the cache was a last minute side effect to the `getConvoUrl`. I can move this out if we don't like side effects.

Reply:
Yea, I'd prefer it to be specifically set

Reply:
So everywhere `getConvoUrl` is called, immediately call `updateConvoUrlCache`?

Reply:
@mwegman  I see that @mattnorris actually requested you to do it this way, so nevermind my request

Reply:
@adamweeks @mwegman we don't have to include this to avoid the side effects. I thought it might make sense to update the cache at this point for convenience, but it may be too magical and go against the grain of a getter.